Assets/LeapMotionModules/InteractionEngine/Scripts/InteractionBehaviours/InteractionGraspDetector.cs

```diff
@@ -0,0 +1,143 @@
+﻿using UnityEngine;
+using System.Collections;
+using Leap.Unity.Attributes;
+using UnityEngine.Events;
+
+namespace Leap.Unity.Interaction {
+  /**
+  * Detects when an interactable object is grasped.
+  * You can specify whether any interactable object or
+  * any object with a specified tag will a activate the detector. You can also
+  * assign a specific list of activating objects.
+  *
+  * All objects must contain an IInteractionBehaviour component and be managed by the
+  * InteractionManager assigned to this detector. 
+  * @since 4.1.5
+  */
+  public class InteractionGraspDetector : Detector {
+
+    /**
+    * The InteractionManager managing the target objects. All
+    * targets must be managed by this manager.
+    * @since 4.1.5
+    */
+    [AutoFind(AutoFindLocations.Scene)]
+    [Tooltip("The Interaction Manager.")]
+    public InteractionManager ieManager = null;
+    /**
+     * The IHandModel instance to observe. 
+     * Set automatically if not explicitly set in the editor.
+     * @since 4.1.5
+     */
+    [AutoFind(AutoFindLocations.Parents)]
+    [Tooltip("The hand model to watch. Set automatically if detector is on a hand.")]
+    public IHandModel HandModel = null;
+
+    /**
+ * The interval at which to check palm direction.
+ * @since 4.1.5
+ */
+    [Tooltip("The interval in seconds at which to check this detector's conditions.")]
+    public float Period = .1f; //seconds
+    /**
+     * Dispatched when a target interactable object is grasped.
+     * The InteractiveBehaviorGraspEvent object provides a reference to the IInteractiveBehavior instance. 
+     * @since 4.1.5
+     */
+    [Tooltip("Dispatched when a target object is grasped.")]
+    public InteractiveBehaviorGraspEvent OnGrasp;
+
+    [Tooltip("Activate when any interactable object is grasped.")]
+    public bool AnyInteractionObject = false;
+    /**
+     * The list of IInteractionBehaviour objects which can activate the detector when held.
+     * Grasping objects in this list will activate the detector even if they do not match the tag
+     * defined in TagName.
+     * @since 4.1.5
+     */
+    [Tooltip("The list of target objects.")]
```

Comment:
I went ahead and added the DisableIf attribute. It doesn't help so much when first setting up the detector, but could help when debugging since you do get an indication that the objects are disabled in the list, if you happen to look.

Also added a slew of attributes to the other Detectors to improve the Inspectors.